Source/mpq/mpq_writer.cpp

```diff
@@ -193,18 +215,12 @@ bool MpqWriter::ReadMPQHeader(_FILEHEADER *hdr)
 	return true;
 }
 
-_BLOCKENTRY *MpqWriter::NewBlock(int *blockIndex)
+MpqBlockEntry *MpqWriter::NewBlock(uint32_t *blockIndex)
 {
-	_BLOCKENTRY *blockEntry = blockTable_;
+	MpqBlockEntry *blockEntry = blockTable_;
 
-	for (int i = 0; i < INDEX_ENTRIES; i++, blockEntry++) {
-		if (blockEntry->offset != 0)
-			continue;
-		if (blockEntry->sizealloc != 0)
-			continue;
-		if (blockEntry->flags != 0)
-			continue;
-		if (blockEntry->sizefile != 0)
+	for (unsigned i = 0; i < BlockEntriesCount; ++i, ++blockEntry) {
+		if (blockEntry->offset != 0 || blockEntry->flags != 0 || blockEntry->unpackedSize != 0 || blockEntry->packedSize != 0)

```

Comment:
This is becoming a bit wide, I think I preferred the multi line version

Reply:
It was a bit odd to have these checks separate. I've now extracted this into a separate function!